### PR TITLE
[Issue:#123] Implement nexus_repository_yum_hosted

### PR DESCRIPTION
**Pull request by ruedigerblock**
This implements the "new" resource nexus_repository_yum_hosted which is derived from resource_repository.go

```
❯ tfa
nexus_repository_yum_hosted.yum: Refreshing state... [id=yummy]
nexus_repository_yum_hosted.yum1: Refreshing state... [id=yummy1]

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

### CODE REVIEW

File: nexus/resource_repository_yum_hosted.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+								"ALLOW",
+								"ALLOW_ONCE",
+								"DENY",
+							}, false),
+						},
+					},
+				},
+			},
+			/* This is remains here until the go client has been adpated */
+			"yum": {
```
Comment (anmoel) [inline nexus/resource_repository_yum_hosted.go:135]: can you handle this in the CRUD functions so it can be deleted?

**Comment by sebbrandt87** [comment]
nice @ruedigerblock 👍 
But did you miss to commit `resource_repository_yum_hosted_test.go` and `data_source_repository_yum_hosted_test.go` ?

File: nexus/data_source_repository_yum_hosted.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+								"ALLOW",
+								"ALLOW_ONCE",
+								"DENY",
+							}, false),
+						},
+					},
+				},
+			},
+			/* This is remains here until the go client has been adpated */
+			"yum": {
```
Comment (anmoel) [inline nexus/data_source_repository_yum_hosted.go:105]: can you handle this in the CRUD functions so it can be deleted?

**Review by anmoel — CHANGES_REQUESTED** [review]

File: nexus/data_source_repository_yum_hosted.go (hunk 2 above)
Comment (ruedigerblock) [reply nexus/data_source_repository_yum_hosted.go:105]: No, I tried ... failed hard.

File: nexus/resource_repository_yum_hosted.go (hunk 1 above)
Comment (ruedigerblock) [reply nexus/resource_repository_yum_hosted.go:135]: No, I tried ... failed hard.

**Comment by ruedigerblock** [comment]
> nice @ruedigerblock 👍 But did you miss to commit ~`resource_repository_yum_hosted_test.go` and~ `data_source_repository_yum_hosted_test.go` or is it not defined yet?

fixed

File: nexus/resource_repository_yum_hosted.go (hunk 1 above)
Comment (ruedigerblock) [reply nexus/resource_repository_yum_hosted.go]: I tried again ... made it.

File: nexus/data_source_repository_yum_hosted.go (hunk 2 above)
Comment (ruedigerblock) [reply nexus/data_source_repository_yum_hosted.go]: I tried again ... made it.

File: nexus/resource_repository_yum_hosted_test.go [hunk 3]
```diff
@@ -28,22 +58,15 @@ func TestAccResourceRepositoryYumHosted(t *testing.T) {
...
-					resourceRepositoryTestCheckFunc(repo),
-					resourceRepositoryTypeHostedTestCheckFunc(repo),
+					resourceYumRepositoryTestCheckFunc(repo),
+					resourceYumRepositoryTypeHostedTestCheckFunc(repo),
 					resource.ComposeAggregateTestCheckFunc(
 						resource.TestCheckResourceAttr(resName, "apt.#", "0"),
 						resource.TestCheckResourceAttr(resName, "bower.#", "0"),
 						resource.TestCheckResourceAttr(resName, "docker.#", "0"),
 						resource.TestCheckResourceAttr(resName, "docker_proxy.#", "0"),
 						resource.TestCheckResourceAttr(resName, "maven.#", "0"),
```
Comment (anmoel) [inline nexus/resource_repository_yum_hosted_test.go:68]: please remove this test. this arguments don't exists in the new resource

File: nexus/resource_repository_yum_hosted_test.go [hunk 4]
```diff
@@ -28,22 +58,15 @@ func TestAccResourceRepositoryYumHosted(t *testing.T) {
...
 					),
-					// Fields related to this format and type
-					// Format
-					resource.ComposeAggregateTestCheckFunc(
-						resource.TestCheckResourceAttr(resName, "yum.#", "1"),
-						resource.TestCheckResourceAttr(resName, "yum.0.deploy_policy", repo.DeployPolicy),
-						resource.TestCheckResourceAttr(resName, "yum.0.repodata_depth", strconv.Itoa(repo.RepodataDepth)),
```
Comment (anmoel) [inline nexus/resource_repository_yum_hosted_test.go:45]: please check this resources on top level

**Review by anmoel — CHANGES_REQUESTED** [review]

File: nexus/resource_repository_yum_hosted_test.go (hunk 3 above)
Comment (anmoel) [reply nexus/resource_repository_yum_hosted_test.go:68]:
```suggestion
					resource.ComposeAggregateTestCheckFunc(
						resource.TestCheckResourceAttr(resName, "deploy_policy", repo.DeployPolicy),
						resource.TestCheckResourceAttr(resName, "repodata_depth", strconv.Itoa(repo.RepodataDepth)),
```

File: nexus/resource_repository_yum_hosted.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+							},
+							Optional: true,
+							Set: func(v interface{}) int {
+								return schema.HashString(strings.ToLower(v.(string)))
+							},
+							Type: schema.TypeSet,
+						},
+					},
+				},
+			},
```
[the remark this answers is not in the document]
Comment (anmoel) [reply nexus/resource_repository_yum_hosted.go:74]: i created a seperated issue

File: docs/resources/repository.md [hunk 6]
```diff
@@ -72,7 +72,7 @@ resource "nexus_repository" "docker_group" {
 
 The following arguments are supported:
 
-* `format` - (Required, ForceNew) Repository format. Possible values: `apt`, `bower`, `conan`, `docker`, `gitlfs`, `go`, `helm`, `maven2`, `npm`, `nuget`, `p2`, `pypi`, `raw`, `rubygems`, `yum`
+* `format` - (Required, ForceNew) Repository format. Possible values: `apt`, `bower`, `conan`, `docker`, `gitlfs`, `go`, `helm`, `maven2`, `npm`, `nuget`, `p2`, `pypi`, `raw`, `rubygems`
```
Comment (anmoel) [inline docs/resources/repository.md:75]:
```suggestion
* `format` - (Required, ForceNew) Repository format. Possible values: `apt`, `bower`, `conan`, `docker`, `gitlfs`, `go`, `helm`, `maven2`, `npm`, `nuget`, `p2`, `pypi`, `raw`, `rubygems`, `yum`
```

File: nexus/resource_repository.go [hunk 7]
```diff
@@ -511,7 +511,15 @@ func resourceRepository() *schema.Resource {
 	}
 }
 
+func RepositoryCleanupDefault() (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (anmoel) [reply nexus/resource_repository.go:514]: i will change it in the next PR

File: nexus/resource_repository_yum_hosted.go [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+							ValidateFunc: validation.StringInSlice([]string{
+								"ALLOW",
+								"ALLOW_ONCE",
+								"DENY",
+							}, false),
+						},
+					},
+				},
+			},
+			"type": {
```
[the remark this answers is not in the document]
Comment (anmoel) [reply nexus/resource_repository_yum_hosted.go:136]: i will change it in the next PR

File: nexus/resource_repository_yum_hosted_test.go (hunk 3 above)
Comment (anmoel) [reply nexus/resource_repository_yum_hosted_test.go:68]: i will change it in the next PR

File: nexus/resource_repository_yum_hosted_test.go (hunk 4 above)
Comment (anmoel) [reply nexus/resource_repository_yum_hosted_test.go:45]: i will change it in the next PR

**Review by anmoel — APPROVED** [review]

**Pull request merged**